### PR TITLE
Feature 2 Course content list on mainpage

### LINKED ISSUE #2: Course content listing
As a user I would like to see the contents of the course.

### PR DESCRIPTION
**Pull request by MrWalrus3451**
Resolves #2

### CODE REVIEW

File: components/screens/MainScreen.tsx [hunk 1]
```diff
@@ -44,8 +66,8 @@ class MainScreen extends React.Component<Props, State> {
     this.state = {
       loading: false,
       error: false,
-      quizData: [],
-      optionsArray: []
+      courseId: 2,
```
Comment (anttileppa) [inline components/screens/MainScreen.tsx:69]: Hard-coded courseId ?

File: components/screens/MainScreen.tsx [hunk 2]
```diff
@@ -62,8 +84,12 @@ class MainScreen extends React.Component<Props, State> {
    * Component did mount lifecycle method
    */
   public async componentDidMount() {
-    this.getTopicsFromMoodle().catch((e) => {
+    this.setState({loading: true});
+    this.getTopicsFromMoodle(this.state.courseId).catch((e) => {
```
Comment (anttileppa) [inline components/screens/MainScreen.tsx]: It's a sync method, why not use await?

File: components/screens/MainScreen.tsx [hunk 3]
```diff
@@ -82,23 +108,86 @@ class MainScreen extends React.Component<Props, State> {
...
-  private async getTopicsFromMoodle() {
-    if (!this.state.moodleToken) {
+  private async getTopicsFromMoodle(courseId: number) {
+    if (!this.props.moodleToken) {
       return this.props.navigation.navigate("Login");
     }
 
-    // TODO Get course topics from moodle and display them
-    const moodleService = Api.getMoodleService("https://ppo-test.metatavu.io", this.state.moodleToken);
+    const moodleService = await Api.getMoodleService("https://ppo-test.metatavu.io", this.props.moodleToken);
```
Comment (anttileppa) [inline components/screens/MainScreen.tsx:139]: Hard-coded url?

File: components/screens/MainScreen.tsx [hunk 4]
```diff
@@ -82,23 +108,86 @@ class MainScreen extends React.Component<Props, State> {
...
+
+    const topics: any = await moodleService.coreCourseGetContents({courseid: courseId});
+
+    const quizService = await Api.getModQuizService("https://ppo-test.metatavu.io", this.props.moodleToken);
```
Comment (anttileppa) [inline components/screens/MainScreen.tsx:143]: See above

File: components/screens/MainScreen.tsx [hunk 5]
```diff
@@ -82,23 +108,86 @@ class MainScreen extends React.Component<Props, State> {
...
+        topicName: topic.name,
+        topicDone: true,
+        topicAvailable: true,
+        topicContent: []
+      }
+      if (topic.modules.length === 0) {
+        newCourseItem.topicDone = false
+      }
+
+      for (const activity of topic.modules) {
```
Comment (anttileppa) [inline components/screens/MainScreen.tsx:161]: const?

File: components/screens/MainScreen.tsx [hunk 6]
```diff
@@ -82,23 +108,86 @@ class MainScreen extends React.Component<Props, State> {
...
+        if (activity.completion === 1 && activity.completiondata.state === 0) {
+          newCourseItem.topicDone = false;
+        }
+        if (activity.modname === "resource") {
+          newCourseItem.topicDone = true;
+        }
+        else if (activity.modname === "quiz") {
```
Comment (anttileppa) [inline components/screens/MainScreen.tsx]: move else if into previous line

File: components/screens/MainScreen.tsx [hunk 7]
```diff
@@ -82,23 +108,86 @@ class MainScreen extends React.Component<Props, State> {
...
+          for (const quiz of quizList.quizzes) {
+            if (quiz.coursemodule === activity.id) {
+              newCourseItem.topicContent.push({name: activity.name, type: "quiz", activityId: quiz.id});
+            }
+          }
+        }
+      }
+      if (newCourseItem.id !== 1) {
```
Comment (anttileppa) [inline components/screens/MainScreen.tsx:176]: Add an empty line before this line

File: components/screens/MainScreen.tsx [hunk 8]
```diff
@@ -82,23 +108,86 @@ class MainScreen extends React.Component<Props, State> {
...
+        courseContent.push(newCourseItem);
+      }
+    }
+    return courseContent;
+  }
+
+  /**
+   * Saves the topic pressed by the user and navigates to the topic page
+   * @param topic
```
Comment (anttileppa) [inline components/screens/MainScreen.tsx]: Describe parameter

**Review by anttileppa — CHANGES_REQUESTED** [review]

File: components/screens/MainScreen.tsx (hunk 1 above)
Comment (MrWalrus3451) [reply components/screens/MainScreen.tsx:69]: Fixed in a later branch

File: components/screens/MainScreen.tsx (hunk 3 above)
Comment (MrWalrus3451) [reply components/screens/MainScreen.tsx:139]: Fixed in a later branch

File: components/screens/MainScreen.tsx (hunk 4 above)
Comment (MrWalrus3451) [reply components/screens/MainScreen.tsx:143]: Fixed in a later branch

File: components/screens/MainScreen.tsx (hunk 5 above)
Comment (MrWalrus3451) [reply components/screens/MainScreen.tsx:161]: const can be used in for of loops if you don't reassign the variable. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of

**Pull request merged**